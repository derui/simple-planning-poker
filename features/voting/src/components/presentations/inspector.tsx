import { Icon, Icons } from "@spp/ui-icon";
import { clsx } from "clsx";

export interface Props {
  name: string;
}

const styles = {
  root: clsx("flex", "flex-col", "p-3", "items-center"),
  name: clsx("flex-auto", "text-indigo-700", "font-bold", "mb-2"),
} as const;

export const Inspector = function Inspector({ name }: Props) {
  return (
    <span className={styles.root}>
      <span className={styles.name}>{name}</span>
      <Icon type={Icons.user} size="xl" variant="indigo" />
    </span>
  );
};
