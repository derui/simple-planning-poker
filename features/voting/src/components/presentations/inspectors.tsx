import { Children, PropsWithChildren } from "react";
import * as styles from "./inspectors.css.js";

export const Inspectors = function Inspectors(props: PropsWithChildren): JSX.Element {
  const childrenCount = Children.count(props.children);
  if (childrenCount == 0) {
    return (
      <div className={styles.root}>
        <p className={styles.emptyText}>No inspectors</p>
      </div>
    );
  }

  return <div className={styles.root}>{props.children}</div>;
};
