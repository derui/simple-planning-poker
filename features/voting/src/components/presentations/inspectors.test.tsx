import { test, afterEach, expect } from "vitest";
import { render, cleanup } from "@testing-library/react";
import { Inspectors } from "./inspectors.js";
import { Inspector } from "./inspector.js";

afterEach(cleanup);

test("should be able to render", () => {
  const ret = render(
    <Inspectors>
      <Inspector name="foo" />
      <Inspector name="bar" />
      <Inspector name="long name" />
    </Inspectors>
  );

  expect(ret.container).toMatchSnapshot();
});

test("render text if no children", () => {
  const ret = render(<Inspectors></Inspectors>);

  expect(ret.container).toMatchSnapshot();
});
