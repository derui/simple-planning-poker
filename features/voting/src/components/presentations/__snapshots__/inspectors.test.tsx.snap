// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render text if no children 1`] = `
<div>
  <div
    class="flex flex-row gap-2 bg-indigo-100 justify-center"
  >
    <p
      class="text-indigo-500 font-bold text-lg m-7"
    >
      No inspectors
    </p>
  </div>
</div>
`;

exports[`should be able to render 1`] = `
<div>
  <div
    class="flex flex-row gap-2 bg-indigo-100 justify-center"
  >
    <span
      class="flex flex-col p-3 items-center"
    >
      <span
        class="flex-auto text-indigo-700 mb-2"
      >
        foo
      </span>
      <span
        class="flex before:bg-indigo-700 before:inline-block before:flex-none before:[mask-image:url(/static/svg/tabler-icons/user.svg)] before:[mask-size:cover] before:[mask-position:center] before:[mask-repeat:no-repeat] before:w-10 before:h-10 w-10 h-10"
      />
    </span>
    <span
      class="flex flex-col p-3 items-center"
    >
      <span
        class="flex-auto text-indigo-700 mb-2"
      >
        bar
      </span>
      <span
        class="flex before:bg-indigo-700 before:inline-block before:flex-none before:[mask-image:url(/static/svg/tabler-icons/user.svg)] before:[mask-size:cover] before:[mask-position:center] before:[mask-repeat:no-repeat] before:w-10 before:h-10 w-10 h-10"
      />
    </span>
    <span
      class="flex flex-col p-3 items-center"
    >
      <span
        class="flex-auto text-indigo-700 mb-2"
      >
        long name
      </span>
      <span
        class="flex before:bg-indigo-700 before:inline-block before:flex-none before:[mask-image:url(/static/svg/tabler-icons/user.svg)] before:[mask-size:cover] before:[mask-position:center] before:[mask-repeat:no-repeat] before:w-10 before:h-10 w-10 h-10"
      />
    </span>
  </div>
</div>
`;
